test/factories/cluster_group.rb

```diff
@@ -0,0 +1,42 @@
+FactoryBot.define do
+  # gets a study object, defaulting to the first user found.  Auto-appends a unique number to the name, and a note
+  # in the description of the study, to aid in test DB uniqueness and cleanup efforts
+  factory :cluster_group do
+    cluster_type { '2d' }
+    cell_annotations { [] }
+    name { study_file.name }
+    study { study_file.study }
+    # create a study but mark as detached, so a Terra workspace is not created
+    factory :cluster_group_with_cells do
+      transient do
+        # cell_data is a hash of three (or 4) arrays: cells, x and y and z
+        cell_data {
+          {
+            x: [1, 2, 3],
+            y: [1, 2, 3],
+            text: ['cellA', 'cellB', 'cellC']
```

Comment:
`cells` seems like the intended field here.

`text` is similar, but as seen [in `load_cluster_group_data_array_points`](https://github.com/broadinstitute/single_cell_portal_core/blob/4661694410edc44c99af0ffde77d153b9a0195b7/app/controllers/site_controller.rb#L1562), it's an array comprised of cell names and other text. `render_cluster` responses reveal it's used only for tooltips in Plotly.  It's also the likeliest candidate for removal from our server-side-constructed Plotly object -- as it's the largest element in the cluster response, and ought to be computable client-side at runtime.  So relying on `cells` would also be more robust to performance-related refactoring.